src/cpp/src/tokenizer.cpp

```diff
@@ -137,10 +137,8 @@ class Tokenizer::TokenizerImpl {
 
     // To change the adding special tokens mode we use a statefull subgraph,
     // this flag holds the current state value of the CompiledModel.
-    bool m_add_special_tokens = true;
-    bool m_skip_special_tokens = true;
-    bool m_pad_to_max_length = false;
-    std::optional<int32_t> m_max_length;
+    ov::AnyMap m_state_flags;
```

Comment:
This solution is more concise and general. We will not need to add new field for every new tokenization parameter!